### PR TITLE
Add slidedown links

### PR DESCRIPTION
**Pull request by franczak**

### CODE REVIEW

File: src/components/Steps/Map/Slidedown.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        title, data, closed, onClick,
+    } = props;
+    return (
+        <div className="slidedown-container">
+            <span
+                onClick={onClick}
+            >
+                {title}
+            </span>
+            <SlideDown closed={closed} className="my-dropdown-slidedown">
```
Comment (chudy1997) [inline src/components/Steps/Map/Slidedown.js]: I think that closed is not really a good, verbose prop... maybe You can pass active or open and just negate it ?

File: src/components/Steps/Map/Slidedown.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+                {data}
+            </SlideDown>
+        </div>
+    );
+};
+
+Slidedown.propTypes = {
+    title: PropTypes.string.isRequired,
+    data: PropTypes.array.isRequired,
```
Comment (chudy1997) [inline src/components/Steps/Map/Slidedown.js]: Is it possible to add some shape of this array ? Do we know what we get here ?

File: src/components/Steps/Map/index.js [hunk 3]
```diff
@@ -6,16 +6,19 @@ import PropTypes from 'prop-types';
...
+const RADIUS = 200;
+const DEFAULT_ZOOM = 10;
 
 class MapStep extends Component {
     constructor(props) {
         super(props);
         this.state = {
             stations: [],
             streetsInAreas: [],
+            closed: [],
```
Comment (chudy1997) [inline src/components/Steps/Map/index.js]: Again isn't it better to hold only index of active one ? If I undestand it right, we have always 1 open, right ?

File: src/components/Steps/Map/index.js [hunk 4]
```diff
@@ -55,11 +60,46 @@ class MapStep extends Component {
         <Station lat={station.coordinates.lat} lng={station.coordinates.lon} />
     ));
 
+    displaySearchLinks = areas => areas.map((area, i) => {
+        const { name } = this.props.city;
+        const data = area.map(street =>
+            (<a
+                href={`https://www.olx.pl/nieruchomosci/mieszkania/${CITIES[name]}/q-${street.split(' ').join('-')}`}
```
Comment (chudy1997) [inline src/components/Steps/Map/index.js]: I'd extract this common part to const.

File: src/components/Steps/Map/index.js [hunk 5]
```diff
@@ -55,11 +60,46 @@ class MapStep extends Component {
...
+                style={{ textDecoration: 'none' }}
```
Comment (chudy1997) [inline src/components/Steps/Map/index.js]: use class instead of inline style.

File: src/components/Steps/Map/index.js [hunk 6]
```diff
@@ -55,11 +60,46 @@ class MapStep extends Component {
...
+                target="_blank"
+            >
+                https://www.olx.pl/nieruchomosci/mieszkania/{CITIES[name]}/q-{street.split(' ').join('-')}
```
Comment (chudy1997) [inline src/components/Steps/Map/index.js]: You use it twice, extract it to const.

File: src/components/Steps/Map/index.js [hunk 7]
```diff
@@ -55,11 +60,46 @@ class MapStep extends Component {
...
+        return (<Slidedown
+            title={`Area  ${i}`}
+            data={data}
+            closed={this.state.closed[i]}
+            onClick={() => this.handleSlideDown(i)}
+        />);
+    });
 
-    displaySearchLinks = areas => areas.map(area => area.map(street => <span>https://www.olx.pl/nieruchomosci/mieszkania/wroclaw/q-{street.split(' ').join('-')}</span>));
+    displayAreas = stations => stations.map((station, index) =>
```
Comment (chudy1997) [inline src/components/Steps/Map/index.js]: Use i or index, please be consistent.

File: src/components/Steps/Map/index.js [hunk 8]
```diff
@@ -55,11 +60,46 @@ class MapStep extends Component {
...
+        (<Area
+            lat={station.coordinates.lat}
+            lng={station.coordinates.lon}
+            importance={index}
+            onClick={() => this.handleSlideDown(index)}
+        />));
+
+    handleSlideDown = (id) => {
+        const closed = this.state.closed.slice(0);
```
Comment (chudy1997) [inline src/components/Steps/Map/index.js]: Again, use active... code will be much cleaner and easier to understand. this is magic for me :P

File: src/const/cities.js [hunk 9]
```diff
@@ -0,0 +1,8 @@
+export default {
+    cities: {
```
Comment (chudy1997) [inline src/const/cities.js]: What about adding this to db and to not worry about it in front ?

File: src/less/components/map.less [hunk 10]
```diff
@@ -25,5 +18,91 @@
...
   }
 }
+
+.my-dropdown-slidedown {
+  display: grid;
+}
+
+.slidedown-container {
+  margin-bottom: 10px;
+  span {
```
Comment (chudy1997) [inline src/less/components/map.less]: What about adding classes to these elements and not use html tags ? Same for a below.

File: src/less/variables.less [hunk 11]
```diff
@@ -4,16 +4,43 @@
...
+
+@bronze: #cd7f32;
+@bronze-hover: #cd7700;
+@bronze-dark: #9c5400;
+
+@orchid: #da70d6;
+@orchid-hover: #b259ae;
+@orchid-dark: #854381;
+
+@pear: #d1e231;
```
Comment (chudy1997) [inline src/less/variables.less]: I admire you for looking for all these values :D Creative man :P

Comment (franczak) [reply src/less/variables.less]: The hardest part of this PR XD

**Pull request merged**